### ISSUE TITLE
Can't perform a query with join on different column names

### ISSUE THREAD

**Issue by lazarevskiVsg** [pola-rs/polars#958]
Hello!

#### Are you using Python or Rust?

**Python**

#### What version of polars are you using?

0.8.9

#### What operating system are you using polars on?

Windows 10

#### Describe your bug.

I'm trying to build an OLAP query engine over polars. 
This can involve many joins, and since I don't see a mechanism to control renaming of duplicate columns, I use column names like "A.col", "B.col", where A and B are tables names.

When an inner join happens on different columns, left column stays, and right column is removed. It's ok for me right now, since I can store a mapping of old columns to a new merged column. 
But the planner panics with an error, caused by trying to refer to removed column.

#### What are the steps to reproduce the behavior?

```python
import polars as pl

cities = pl.DataFrame({
    "Cities.City": ["Moscow", "Berlin", "Paris","Hamburg", "Lyon", "Novosibirsk"],
    "Cities.Population": [11.92, 3.645, 2.161, 1.841, 0.513, 1.511],
    "Cities.Country": ["Russia", "Germany", "France", "Germany", "France", "Russia"]
})

print(cities)

sales = pl.DataFrame({
    "Sales.City": ["Moscow", "Berlin", "Paris", "Moscow", "Berlin", "Paris", "Moscow", "Berlin", "Paris"],
    "Sales.Item": ["Item A", "Item A","Item A",
                   "Item B", "Item B","Item B",
                   "Item C", "Item C","Item C"],
    "Sales.Amount": [200, 180, 100, 
                    3, 30, 20,
                    90, 130, 125]
})

print(sales)

print(sales.lazy().join(cities.lazy(),left_on="Sales.City",right_on="Cities.City").collect())

result = sales.lazy().join(cities.lazy(),left_on="Sales.City",right_on="Cities.City")\
  .groupby(["Cities.Country"]).agg([pl.col("Sales.Amount").apply(lambda x: x.sum())]).collect()
print(result)
```

#### What is the actual behavior?
There is a final query plan:
```
Aggregate
	[APPLY([Sales.Amount])] BY [Cities.Country] FROM JOIN
	(TABLE: ["Sales.City", "Sales.Item", "Sales.Amount"]; PROJECT */3 COLUMNS; SELECTION: None)
WITH
	(TABLE: ["Cities.City", "Cities.Population", "Cities.Country"]; PROJECT */3 COLUMNS; SELECTION: None)
ON (left: [Sales.City] right: [Cities.City])
```

And there is the error
```
PanicException                            Traceback (most recent call last)
<ipython-input-19-cc82baa24399> in <module>
----> 1 result = sales.lazy().join(cities.lazy(),left_on="Sales.City",right_on="Cities.City")\
      2     .groupby(["Cities.Country"]).agg([pl.col("Sales.Amount").apply(lambda x: x.sum())]).collect()
      3 print(result)

venv\lib\site-packages\polars\lazy\__init__.py in collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, string_cache, no_optimization)
    340             string_cache,
    341         )
--> 342         return wrap_df(ldf.collect())
    343 
    344     def fetch(

PanicException: called `Result::unwrap()` on an `Err` value: NotFound("Unable to get field named \"Cities.City\". Valid fields: [\"Sales.City\", \"Sales.Item\", \"Sales.Amount\", \"Cities.Population\", \"Cities.Country\"]")
```

#### What is the expected behavior?
I have run the same thing with pandas:

```python
sp = sales.to_pandas()
cp = cities.to_pandas()
print(sp.merge(cp,left_on="Sales.City", right_on="Cities.City")\
      .groupby("Cities.Country").agg({"Sales.Amount":lambda x:x.sum()}))
```

```
                Sales.Amount
Cities.Country              
France                   245
Germany                  340
Russia                   293
```
      
Column "Cities.City" also stays in merged dataframe, unlike in polars, where it is deleted, though I'm not sure, which behavior is better.

Thank you.

**Comment by ritchie46** [comment]
Thanks for the issue report. This is a bug in the `projection pushdown` optimization. I will fix this.

You can run youre query for the time being by disabling projection pushdown for that query: `.collect(projection_pushdown=False)`.

P.S. there is a sum expression. :)

```python
result = (sales.lazy()
          .join(cities.lazy(), left_on="Sales.City", right_on="Cities.City")
          .groupby(["Cities.Country"])
          .agg([pl.col("Sales.Amount").sum()])
          .collect(projection_pushdown=False))
```

**Comment by ritchie46** [comment]
This is fixed in #959. You can compile from source following the steps described in the readme. 

Or you can wait a bit and then it will be fixed in the release of next friday. :)
